lib/test_runner.rb

```diff
@@ -116,7 +141,29 @@ def run_tests(tests, args:)
       begin
         tests[:setup]&.run!(args: args)
         tests[:tap_syntax]&.run!(args: args)
-        tests[:formulae]&.run!(args: args)
+
+        formulae, added_formulae, deleted_formulae, built_formulae, test_default_formula = nil
```

Comment:
Perhaps initially set these to `[]` instead so we don't need `||= []` everywhere.

`test_default_formula` can be set from `args` right away. I'd possibly even pass it in to the constructor instead of passing it around here.